shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java

```diff
@@ -0,0 +1,117 @@
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class ShardingRenameTableStatementValidatorTest {
+
+    @Mock
+    private ShardingRule shardingRule;
+
+    @Test(expected = ShardingSphereException.class)
+    public void assertPreValidateShardingTable() {
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_order", "t_user_order");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        when(shardingRule.tableRuleExists(argThat(tableNames -> tableNames.contains("t_order") || tableNames.contains("t_user_order")))).thenReturn(true);
+        new ShardingRenameTableStatementValidator().preValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema);
+    }
+
+    @Test(expected = ShardingSphereException.class)
+    public void assertPreValidateBroadcastTable() {
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_order", "t_user_order");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        when(shardingRule.isBroadcastTable(eq("t_order"))).thenReturn(true);
+        new ShardingRenameTableStatementValidator().preValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema);
+    }
+
+    @Test
+    public void assertPreValidateNormalCase() {
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_not_sharding_table", "t_not_sharding_table_new");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        new ShardingRenameTableStatementValidator().preValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema);
+    }
+
+    @Test(expected = ShardingSphereException.class)
+    public void assertPostValidateDifferentRouteUnitsAndDataNodesSize() {
+        // route units and actual data nodes are different size
+        RouteContext routeContext = new RouteContext();
+        routeContext.getRouteUnits().add(mock(RouteUnit.class));
+        TableRule tableRule = mock(TableRule.class);
+        when(tableRule.getActualDataNodes()).thenReturn(Arrays.asList(mock(DataNode.class), mock(DataNode.class)));
+        when(shardingRule.getTableRule("t_order")).thenReturn(tableRule);
+        when(shardingRule.isShardingTable("t_order")).thenReturn(true);
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_order", "t_user_order");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        ConfigurationProperties props = mock(ConfigurationProperties.class);
+        new ShardingRenameTableStatementValidator().postValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema, props, routeContext);
+    }
+
+    @Test
+    public void assertPostValidateNormalCase() {
+        // route units and actual data nodes are equal size
```

Comment:
@homeway88 Please remove this comment here.

Reply:
Done